From ac596ca1c13faf9739e1b835ea86dc4200d09b27 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marco=20Trevisan=20=28Trevi=C3=B1o=29?= <mail@3v1n0.net>
Date: Tue, 5 Dec 2017 17:04:24 -0500
Subject: [PATCH 07/12] dash: Do not shadow ClutterActor's destroy()

Since commit ef1e27966db2 turned DashItemContainer into an StWidget,
the destroy() method overrides the ClutterActor method, which is at
the very least bad style. Instead, follow the usual pattern of using
a ::destroy handler.

https://bugzilla.gnome.org/show_bug.cgi?id=791233
---
 js/ui/dash.js | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/js/ui/dash.js b/js/ui/dash.js
index b89040bb6..43b7c4450 100644
--- a/js/ui/dash.js
+++ b/js/ui/dash.js
@@ -50,6 +50,10 @@ var DashItemContainer = new Lang.Class({
         this._childScale = 0;
         this._childOpacity = 0;
         this.animatingOut = false;
+
+        this.connect('destroy', () => {
+            this.label.destroy();
+        });
     },
 
     vfunc_allocate: function(box, flags) {
@@ -177,11 +181,6 @@ var DashItemContainer = new Lang.Class({
                          });
     },
 
-    destroy: function() {
-        this.label.destroy();
-        this.parent();
-    },
-
     animateOutAndDestroy: function() {
         this.label.hide();
 
-- 
2.15.1

